pygit2/__init__.py

```diff
@@ -61,6 +61,8 @@
 from .utils import to_bytes, to_str
 
 
+__all__ = ('Repository',)
```

Comment:
There is more in pygit2 than `Repository`. Please drop this change from this PR and I will merge it, and then maybe open a new PR to define `__all__`.

Reply:
Sure. 
I'm not that familar with pygit2 - what else should be added to `__all__`? The only issue I've met was related with `Repository`. Though I think submitting this change I've missed that `__all__` may also affect the users that rely on `from pygit2 import *`.

If it's not clear at the moment what should go to `__all__`, then maybe we can hide it under `TYPE_CHECKING` to avoid any runtime implications?

```python
if typing.TYPE_CHECKING:
    __all__ = ("Repository",)
```

Reply:
Created a separate PR https://github.com/libgit2/pygit2/pull/1371